src/main/java/cubicchunks/world/cube/Cube.java

```diff
@@ -422,6 +429,73 @@ public void randomTick(WorldServer worldServer, Random rand) {
             block.randomTick(worldServer, pos, iblockstate, rand);
         }
     }
+    
+    public void updateThunder(WorldServer worldServer, Random rand, boolean doMobSpawning) {
+        this.updateLCG = this.updateLCG * 3 + 1013904223;
+        int j1 = updateLCG >> 2;
+        int localX = AddressTools.getLocalX(j1);
+        int localZ = AddressTools.getLocalZ(j1);
+        int localY = AddressTools.getLocalY(j1);
+        BlockPos randomPos = this.coords.localToBlock(localX, localY, localZ);
+        BlockPos strikeTarget = this.column.getPrecipitationHeight(randomPos);
+        if (Coords.blockToCube(strikeTarget.getY()) != this.getY())
```

Comment:
it doesn't really make sense to do this in cube. Instead this should be done in column with isCubeLoaded check. This would save a lot of iterations if done right (also `O(R^2)` instead of `O(R^3)` for R=render distance)